src/utils.jl

```diff
@@ -60,4 +62,15 @@ function CRC.rrule(::typeof(_iqf), x)
 
     y, ∇_iqf
 end
-#
+
+#======================================================#
+# Batched Multiplication using Einstein Summation
+#======================================================#
+
+@inline function batched_mul(x, coeffs)
```

Comment:
Can you not use NNlib.batched_mul or NNlib.batched_vec for this?

Reply:
https://fluxml.ai/NNlib.jl/dev/reference/#NNlib.batched_mul

Reply:
I am not sure. I have been trying but I have not able to get it right